libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewListener.kt

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.Style
+import com.mapbox.navigation.dropin.component.camera.TargetCameraMode
+import com.mapbox.navigation.dropin.component.destination.Destination
+import com.mapbox.navigation.dropin.component.navigation.NavigationState
+
+/**
+ * Interface definition for the NavigationView.
+ */
+abstract class NavigationViewListener {
+    /**
+     * Called when [Destination] has changed. Might be called multiple times once
+     * [Destination.features] are updated.
+     *
+     * @param destination New destination value.
+     */
+    open fun onDestinationChanged(destination: Destination?) = Unit
+
+    /**
+     * Called when [NavigationState] has changed.
+     *
+     * @param state New navigation state value.
+     */
+    open fun onNavigationStateChanged(state: NavigationState) = Unit
+
+    /**
+     * Called when Map [Style] has changed. Invoked once the new style has been fully loaded,
+     * including the style specified sprite and sources.
+     *
+     * @param style Fully loaded style.
+     */
+    open fun onMapStyleChanged(style: Style) = Unit
+
+    /**
+     * Called when [TargetCameraMode] has changed.
+     *
+     * @param mode New camera mode value.
+     */
+    open fun onCameraModeChanged(mode: TargetCameraMode) = Unit
+
+    /**
+     * Called when NavigationCamera padding has changed.
+     *
+     * @param padding New camera padding value.
+     */
+    open fun onCameraPaddingChanged(padding: EdgeInsets) = Unit
+
+    /**
+     * Called when Audio Guidance mute state has change.
+     *
+     * @param muted Audio Guidance muted state.
+     */
+    open fun onAudioGuidanceStateChanged(muted: Boolean) = Unit
+
+    /**
+     * Called when Camera Recenter button has been clicked. This allows listeners to get a chance
+     * to respond before the NavigationView.
+     *
+     * @return `True` if the listener has consumed the event, otherwise `false`.
+     */
+    open fun onRecenterClick(): Boolean = false
```

Comment:
NIT: Based on other methods above, and the fact that these are observers, we should probably use past tense `onRecenterClicked`. Similar for other clicks below.